packages/edit-post/src/components/layout/index.js

```diff
@@ -359,12 +408,7 @@ function Layout( {
 					}
 					extraContent={
 						! isDistractionFree &&
-						showMetaBoxes && (
-							<div className="edit-post-layout__metaboxes">
-								<MetaBoxes location="normal" />
-								<MetaBoxes location="advanced" />
-							</div>
-						)
+						showMetaBoxes && <MetaBoxesWide />
```

Comment:
```suggestion
						showMetaBoxes &&
						hasActiveMetaboxes && <MetaBoxesWide />
```

I think if we don't check if there are active metaboxes the resizable box will always be rendered.

Reply:
Good catch! I tried the suggestion but `hasActiveMetaboxes` is not sufficient either as it doesn’t account for their visible status.
<details><summary>I found a way using `getMetaBoxesPerLocation` and `isEditorPanelEnabled` but it seems a little heavy and is redundant with what child components `Metaboxes`/`MetaboxVisibility` are doing.</summary>

```diff
diff --git a/packages/edit-post/src/components/layout/index.js b/packages/edit-post/src/components/layout/index.js
index 3919cd0d6a..2627f4c618 100644
--- a/packages/edit-post/src/components/layout/index.js
+++ b/packages/edit-post/src/components/layout/index.js
@@ -229,6 +229,7 @@ function Layout( {
 		mode,
 		isFullscreenActive,
 		hasActiveMetaboxes,
+		isAnyMetaboxEnabled,
 		hasBlockSelected,
 		showIconLabels,
 		isDistractionFree,
@@ -252,12 +253,20 @@ function Layout( {
 				kind: 'postType',
 				name: 'wp_template',
 			} );
+			const { getMetaBoxesPerLocation } = select( editPostStore );
+			const { isEditorPanelEnabled } = select( editorStore );
 
 			return {
 				mode: select( editorStore ).getEditorMode(),
 				isFullscreenActive:
 					select( editPostStore ).isFeatureActive( 'fullscreenMode' ),
 				hasActiveMetaboxes: select( editPostStore ).hasMetaBoxes(),
+				isAnyMetaboxEnabled: [
+					...getMetaBoxesPerLocation( 'normal' ),
+					...getMetaBoxesPerLocation( 'advanced' ),
+				].some( ( { id } ) =>
+					isEditorPanelEnabled( `meta-box-${ id }` )
+				),
 				hasBlockSelected:
 					!! select( blockEditorStore ).getBlockSelectionStart(),
 				showIconLabels: get( 'core', 'showIconLabels' ),
@@ -408,7 +417,8 @@ function Layout( {
 					}
 					extraContent={
 						! isDistractionFree &&
-						showMetaBoxes && <MetaBoxesWide />
+						showMetaBoxes &&
+						isAnyMetaboxEnabled && <MetaBoxesWide />
 					}
 				>
 					<PostLockedModal />

```
</details>

So I tried with CSS instead c0d591649bc5a89862ff9967cd73f5761bc803ef. I prefer it a little (grudgingly).